### PR TITLE
1228 support edit accrediting provider 

### PR DESCRIPTION
**Pull request by goodviber**
### Context
User need

As a user, I want to easily add accrediting providers to lead schools, so that these providers can be setup correctly in Publish
Why

This will give users and providers flexibility in managing their APs.
What needs to be done

We need to implement the edit functionality for AP in the new AP tab for lead schools so the description can be changed.
Done when

This is our definition of done

We are able to edit an AP’s description from the lead school's AP tab

### Changes proposed in this pull request
Update routes/form/controller to allow update of provider.accredited_provider_enrichments.

### Guidance to review
In support console, select a  provider with accredited providers, click 'Add accredited provider' and update description via Change link
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [ ] Inform data insights team due to database changes

### CODE REVIEW

File: app/forms/support/accredited_provider_form.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Support
+  class AccreditedProviderForm < Form
```
Comment (darokel) [inline app/forms/support/accredited_provider_form.rb:4]: I removed this in an earlier PR as I feel we should just be using the same form for support/publish. It's essentially the same behaviour? Couldn't we use the existing one https://github.com/DFE-Digital/publish-teacher-training/blob/main/app/forms/accredited_provider_form.rb

Comment (goodviber) [reply app/forms/support/accredited_provider_form.rb]: Done

File: config/locales/en.yml [hunk 2]
```diff
@@ -346,7 +346,12 @@ en:
         new:
           title: About the accredited provider
           hint: Tell candidates about the accredited provider. You could mention their academic specialities and achievements.
-          caption: *accredited_provider_caption
+          caption: "Add accredited provider - %{provider_name} (%{code})"
+        edit: 
+          title: About the accredited provider
+          hint: Tell candidates about the accredited provider. You could mention their academic specialities and achievements.
+          caption: "%{provider_name}"
+          updated: About the accredited provider updated
```
Comment (defong) [inline config/locales/en.yml:354]: Are you actually using any of these?

Comment (goodviber) [reply config/locales/en.yml:354]: Screenshot 2023-05-19 at 14 46 00

Yes, I think so.

Comment (defong) [reply config/locales/en.yml:354]: Whats with caption?

**Pull request merged**